plugins/out_oracle_log_analytics/oci_logan.h

```diff
@@ -0,0 +1,223 @@
+#define FLB_OCI_PARAM_COLLECT_TIME_FIELD "collect_time_field_name"
+
+#define FLB_OCI_PARAM_USE_RAW_RECORD "use_raw_record"
+#define FLB_OCI_PARAM_USE_RAW_RECORD_SIZE sizeof(FLB_OCI_PARAM_USE_RAW_RECORD)-1
+
+#define FLB_OCI_PARAM_INCLUDE_COLLECT_TIME "include_collect_time"
+#define FLB_OCI_PARAM_INCLUDE_COLLECT_TIME_SIZE sizeof(FLB_OCI_PARAM_INCLUDE_COLLECT_TIME)-1
+
+#define FLB_OCI_MATCH_ID_MAX 1000 // TO avoid too large memory allocation
+
+#define FLB_OCI_DEFAULT_COLLECT_TIME       "oci_collect_time"
+#define FLB_OCI_DEFAULT_COLLECT_TIME_SIZE sizeof(FLB_OCI_DEFAULT_COLLECT_TIME)-1
+
+/* Http Header */
+#define FLB_OCI_HEADER_REQUEST_TARGET           "(request-target)"
+#define FLB_OCI_HEADER_USER_AGENT                      "User-Agent"
+#define FLB_OCI_HEADER_USER_AGENT_VAL                  "Fluent-Bit"
+#define FLB_OCI_HEADER_CONTENT_TYPE                    "content-type"
+#define FLB_OCI_HEADER_CONTENT_TYPE_VAL                "application/octet-stream"
+#define FLB_OCI_HEADER_X_CONTENT_SHA256                "x-content-sha256"
+#define FLB_OCI_HEADER_CONTENT_LENGTH                  "content-length"
+#define FLB_OCI_HEADER_HOST                            "host"
+#define FLB_OCI_HEADER_DATE                            "date"
+#define FLB_OCI_HEADER_AUTH                            "Authorization"
+#define FLB_OCI_PAYLOAD_TYPE                           "payloadType"
+
+
+/* For OCI signing */
+#define FLB_OCI_PARAM_TENANCY     "tenancy"
+#define FLB_OCI_PARAM_USER        "user"
+#define FLB_OCI_PARAM_KEY_FINGERPRINT     "fingerprint"
+#define FLB_OCI_PARAM_KEY_FILE     "key_file"
+#define FLB_OCI_PARAM_REGION  "region"
+#define FLB_OCI_PARAM_KEY_FILE_PASSPHRASE "key_file_passphrase"
+
+#define FLB_OCI_SIGN_SIGNATURE_VERSION   "Signature version=\"1\""
+#define FLB_OCI_SIGN_KEYID   "keyId"
+#define FLB_OCI_SIGN_ALGORITHM   "algorithm=\"rsa-sha256\""
+
+#define FLB_OCI_SIGN_HEADERS     "headers=\"" \
+    FLB_OCI_HEADER_REQUEST_TARGET " " \
+    FLB_OCI_HEADER_HOST " " \
+    FLB_OCI_HEADER_DATE " " \
+    FLB_OCI_HEADER_X_CONTENT_SHA256 " " \
+    FLB_OCI_HEADER_CONTENT_TYPE " " \
+    FLB_OCI_HEADER_CONTENT_LENGTH "\""
+
+#define FLB_OCI_SIGN_SIGNATURE   "signature"
+
+/* For error response */
+#define FLB_OCI_ERROR_RESPONSE_CODE     "code"
+#define FLB_OCI_ERROR_RESPONSE_MESSAGE  "message"
+
+#define FLB_OCI_ERROR_CODE_RELATED_RESOURCE_NOT_FOUND      "RelatedResourceNotAuthorizedOrNotFound"
+#define FLB_OCI_ERROR_CODE_NOT_AUTHENTICATED               "NotAuthenticated"
+#define FLB_OCI_ERROR_CODE_NOT_AUTHENTICATEDORNOTFOUND     "NotAuthorizedOrNotFound"
+#define FLB_OCI_ERROR_CODE_INCORRECTSTATE                  "IncorrectState"
+#define FLB_OCI_ERROR_CODE_NOT_AUTH_OR_RESOURCE_EXIST      "NotAuthorizedOrResourceAlreadyExists"
+#define FLB_OCI_ERROR_CODE_TOO_MANY_REQUESTS               "TooManyRequests"
+#define FLB_OCI_ERROR_CODE_INTERNAL_SERVER_ERROR           "InternalServerError"
+
+#define METADATA_URL_BASE  "http://169.254.169.254/opc/v2"
```

Comment:
why these endpoints are hard coded and there is no host/DNS ? are these definitions in use ?

Reply:
These are not in use currently but will be in use in the follow up PR to add two other modes of authentication to connect to Oracle Cloud Infrastructure endpoints. I can remove it from this PR if you want.

The reason for no host/DNS is probably because these endpoints were set this way. OCI has SDKs in various languages and they include these endpoints hardcoded just like this.

Reply:
Removed those endpoints from this PR since they are unused currently.